File: cpp/UVA100.cpp
//https://zerojudge.tw/ShowProblem?problemid=c039
#include <iostream>
using namespace std;

int main()
{
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    int i, j;
    while (cin >> i >> j)
    {
        int mx = 0;
        for (int n = min(i, j); n <= max(i, j); n++)
        {
            int n1 = n, cnt = 1;
            while (n1 != 1)
            {
                if (n1 % 2 == 1)
                    n1 = 3 * n1 + 1;
                else
                    n1 /= 2;
                cnt++;
            }
            mx = max(mx, cnt);
        }
        cout << i << " " << j << " " << mx << "\n";
    }
    return 0;
}

File: java/UVA100.java
//https://zerojudge.tw/ShowProblem?problemid=c039
import java.util.*;

public class UVA00100 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        List<String> results = new ArrayList<>();
        while (sc.hasNextInt()) {
            int i = sc.nextInt();
            int j = sc.nextInt();
            int from = Math.min(i, j);
            int to = Math.max(i, j);
            int maxCycle = 0;
            for (int n = from; n <= to; n++) {
                int count = 1;
                int temp = n;
                while (temp != 1) {
                    if (temp % 2 == 1) {
                        temp = 3 * temp + 1;
                    } else {
                        temp = temp / 2;
                    }
                    count++;
                }
                if (count > maxCycle) {
                    maxCycle = count;
                }
            }
            results.add(String.format("%d %d %d", i, j, maxCycle));
        }
        for (String res : results) {
            System.out.println(res);
        }
    }
}
